2.cellprofiler_processing/cputils.py

```diff
@@ -8,18 +8,22 @@
 from pathlib import Path
 
 
-def run_cellprofiler(path_to_pipeline: str, path_to_output: str, path_to_images: str):
+def run_cellprofiler(path_to_pipeline: str, path_to_output: str, path_to_images: str, plate_name: str):
     """Profile batch with CellProfiler (runs segmentation and feature extraction)
 
     Args:
         path_to_pipeline (str): path to the CellProfiler .cppipe file with the segmentation and feature measurement modules
         path_to_output (str): path to the output folder for the .sqlite file
         path_to_images (str): path to folder with IC images from specific plate
     """
-    # run CellProfiler on a plate
-    command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}"
-    os.system(command)
-
+    # runs through any files that are in the output path
+    if any(files.name.startswith(plate_name) for files in pathlib.Path(path_to_output).iterdir()):
+        print("This plate has already been analyzed!")
+        return
+    else:
+        # run CellProfiler on a plate
+        command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}"
+        os.system(command)
```

Comment:
Consider removing the `else:` line for this if block (and unindenting) to simplify and increase readability.

```suggestion
    # run CellProfiler on a plate
    command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}"
    os.system(command)
```

Reply:
👍